src/Terminal/commands/grep.ts

```diff
@@ -163,24 +160,26 @@ class Args {
   }
 
   private spliceOptionalParams(): Args {
-    for (const [key, args] of Object.keys(this.params).map((key): [string, ArgStrings] => [
-      key,
-      VALID_ARGS[key as keyof Options],
-    ])) {
-      this.params[key as keyof Params] = this.spliceParam(args);
+    for (const [key, validArgs] of Object.entries(VALID_PARAMS)) {
+      this.params[key as keyof Parameters] = this.spliceParam(validArgs);
     }
     return this;
   }
 
   private reduceToOptionsAndFiles(): string[] {
     const stripDash = (arg: string) => arg.slice(1);
     const argKeys = Object.keys(VALID_ARGS).map((k) => k as keyof Options);
+    const paramKeys = Object.keys(VALID_PARAMS).map((k) => k as keyof Parameters);
```

Comment:
What do these maps do? They look like identity maps (i.e. no-ops) to me.

You're just using these to iterate over VALID_PARAMS/VALID_ARGS, and you're looking up the entry there as well. It would be better to `for (const value of Object.values(VALID_ARGS))` down below, instead.